primitives/src/targeting/eval.rs

```diff
@@ -12,12 +12,46 @@ pub type Map = serde_json::value::Map<String, SerdeValue>;
 
 use super::{Input, Output};
 
+#[cfg(test)]
+#[path = "eval_test.rs"]
+mod test;
+
 #[derive(Debug, Eq, PartialEq)]
 pub enum Error {
     TypeError,
     UnknownVariable,
 }
 
+fn get_deposit_asset_divisor(address: &String) -> Result<BigNum, Error> {
```

Comment:
`&String` should be `&str`, it's always better to have the slices (&str, &[Type]) instead of the owned values (&String, Vec<Type>).